src/deluge/gui/views/session_view.cpp

```diff
@@ -3525,15 +3548,18 @@ ActionResult SessionView::gridHandlePadsLaunchWithSelection(int32_t x, int32_t y
 
 void SessionView::gridHandlePadsLaunchToggleArming(Clip* clip, bool immediate) {
 	if (immediate) {
-		if (currentUIMode == UI_MODE_HOLDING_HORIZONTAL_ENCODER_BUTTON) {
+		if (horizontalEncoderPressed) {
```

Comment:
shouldn't this be `isUIModeActive(UI_MODE_HOLDING_HORIZONTAL_ENCODER_BUTTON)`, rather than recreating that tracking with `horiznotalEncoderPressed`?

Reply:
Normally yes, that's how it was before but it got overwritten by clip being held state. Unfortunately there are a lot of places where ui state is written exclusively and sometimes the behavior relies on previous state to be overwritten and sometimes on an added state in some paths and overwritten in others, etc. The whole session view needs to be overhauled to be consistent again and to make future changes easier.

I hope this little hack to not break anything before Release is justified :D.

Reply:
```
$ rg 'currentUIMode = ' | wc -l
112
```
yeah that's probably too much to deal with for now